dht/dht.go

```diff
@@ -49,24 +49,39 @@ type Server struct {
 	socket           net.PacketConn
 	transactions     map[transactionKey]*Transaction
 	transactionIDInt uint64
-	nodes            map[string]*node // Keyed by dHTAddr.String().
+	nodes            map[string]*Node // Keyed by dHTAddr.String().
 	mu               sync.Mutex
 	closed           chan struct{}
 	ipBlockList      iplist.Ranger
 	badNodes         *boom.BloomFilter
+	hooks            map[string]KRPCHook
 
 	numConfirmedAnnounces int
 	bootstrapNodes        []string
 	config                ServerConfig
 }
 
+// Returned when a caller sets ID manually but fails to set NoSecurity
+// configuration flag. Better to be explicit and make outcomes
+// predictable than implicitly disable one option when provided another.
+var ErrConflictingConfigNoSecNodeId = errors.New("Cannot manually set NodeId without also setting NoSecurity.")
+
 type ServerConfig struct {
 	Addr string // Listen address. Used if Conn is nil.
-	Conn net.PacketConn
+	// To set NodeId manually. Caller MUST also set NoSecurity,
+	// and NodeId in this case is 20 bytes cast to string, not
+	// a hex or baseXX encoded ID.
+	NodeId string
+	Conn   net.PacketConn
 	// Don't respond to queries from other nodes.
 	Passive bool
 	// DHT Bootstrap nodes
 	BootstrapNodes []string
+	// Disable bootstrapping from global servers even if given no BootstrapNodes.
+	// This creates a solitary node that awaits other nodes; it's only useful if
+	// you're creating your own DHT and want to avoid accidental crossover, without
+	// spoofing a bootstrap node and filling your logs with connection errors.
+	NoBootstrap bool
```

Comment:
Looks good, I assume I use the default bootstrap nodes if none are provided in BootstrapNodes, which is bad style.